src/hooks/UseRepositories.ts

```diff
@@ -13,35 +13,72 @@ export function useRepositories(organization?: string) {
 
   // Keep state of current search in this hook
   const [page, setPage] = useState(1);
+  const [nextPageToken, setNextPageToken] = useState('');
   const [perPage, setPerPage] = useState(10);
+  const [repos, setRepos] = useState([]);
   const [search, setSearch] = useState<SearchState>({
     field: ColumnNames.name,
     query: '',
   });
   const [currentOrganization, setCurrentOrganization] = useState(organization);
 
+  const setNextPageParams = (nextPageToken) => {
+    setNextPageToken(nextPageToken || '');
+    setPage(page + 1);
+  };
+
   const listOfOrgNames: string[] = currentOrganization
     ? [currentOrganization]
     : user?.organizations.map((org) => org.name).concat(user.username);
 
   const {
-    data: repositories,
+    data,
     isLoading: loading,
     isPlaceholderData,
     error,
-  } = useQuery(
-    ['organization', organization, 'repositories'],
-    currentOrganization
-      ? ({signal}) => fetchRepositoriesForNamespace(currentOrganization, signal)
-      : ({signal}) => fetchAllRepos(listOfOrgNames, true, signal),
-    {
-      placeholderData: [],
+  } = useQuery({
```

Comment:
It seems there are two paging approaches conflated in this implementation. As of now when clicking through the pages in the `projectquay` organization all 150 repos are being appended to the total list. 
![image](https://user-images.githubusercontent.com/22058392/211582536-083a04a0-1109-4305-9f30-269669dbbc40.png)
Gathering all the repos up front has already been implemented in RepositoryResource.ts:L81. The changes here should probably be reverted in favor of the other approach.

Reply:
Realised that [RepositoryResource.ts:L81.](https://github.com/quay/quay-ui/pull/166/files#diff-281eaf07bf2aa85cb2f5c77cb7478e221613871ced40e342bfb6e872b5d574feR81) actually needs to be an if condition. I'd need the `onSuccess` function to handle the different types of responses returned.